src/http_rest_api_handler.cpp

```diff
@@ -497,6 +498,33 @@ Status HttpRestApiHandler::prepareGrpcRequest(const std::string modelName, const
     return StatusCode::OK;
 }
 
+static std::set<std::string> getRequestedBinaryOutputs(::KFSRequest& grpc_request, ::KFSResponse& grpc_response) {
+    std::set<std::string> binaryOutputs;
+    bool allBinaryOutputsRequested = false;
+    for (auto parameter : grpc_request.parameters()) {
+        if (parameter.second.parameter_choice_case(), inference::InferParameter::ParameterChoiceCase::kBoolParam) {
+            if (parameter.first == "binary_data_output") {
+                allBinaryOutputsRequested = parameter.second.bool_param();
+                break;
+            }
+        }
+    }
+    for (const inference::ModelInferRequest_InferRequestedOutputTensor output : grpc_request.outputs()) {
+        bool specificBinaryOutputRequested = false;
+        for (auto parameter : output.parameters()) {
+            if ((parameter.second.parameter_choice_case() == inference::InferParameter::ParameterChoiceCase::kBoolParam) &&
+                (parameter.first == "binary_data")) {
+                specificBinaryOutputRequested = parameter.second.bool_param();
+                break;
+            }
+        }
+        if (specificBinaryOutputRequested || allBinaryOutputsRequested) {
+            binaryOutputs.insert(output.name());
```

Comment:
We need to look up outputs parameters because the binary_data output parameter overrides binary_data_output request parameter.  Although there was another problem, already solved.

Reply:
maybe rename allBinaryOutputsRequested to-> byDefaultBinaryOutpuRequested?